pinot/representation/sequential.py

```diff
@@ -57,27 +49,34 @@ def _fn(g):
             elif isinstance(exe, str):
                 activation = getattr(torch.nn.functional, exe)
 
-                setattr(self, "a" + str(idx), apply_atom_in_graph(activation))
+                setattr(self, "a" + str(idx), activation)
 
                 self.exes.append("a" + str(idx))
 
             # float -> dropout
             elif isinstance(exe, float):
                 dropout = torch.nn.Dropout(exe)
-                setattr(self, "o" + str(idx), apply_atom_in_graph(dropout))
+                setattr(self, "o" + str(idx), dropout)
 
                 self.exes.append("o" + str(idx))
 
-    def forward(self, g, return_graph=False):
+    def forward(self, g, x=None, 
+            pool=lambda g: dgl.sum_nodes(g, 'h')):
+    
+        if x is None:
+            x = g.ndata['h']
 
-        g.apply_nodes(lambda nodes: {"h": self.f_in(nodes.data["h0"])})
+        x = self.f_in(x)
```

Comment:
Would this do in place mopdifications by overwriting x because x now points to g.ndata['h']?

Reply:
@yuanqing-wang

Reply:
no it wouldn't

both g.ndata['h'] and x are simple tensor objects and only underscore methods can do in-place modification